force-app/main/default/classes/WeatherIntegrationControllerExtension.cls

```diff
@@ -0,0 +1,38 @@
+public with sharing class WeatherIntegrationControllerExtension {
+    
+	private Account accountRecord {
+        get{
+            if(accountRecord == null){
+                accountRecord = (Account) sdtController.getRecord();
+            }
+            return accountRecord;
+        }
+        set;
+    }
+    public Wrapperclass wrapper {get; set;}
+    public string urlIcon {get; set;}
+    
+    private ApexPages.StandardController sdtController;
+    private static final string API_KEY = '592da47fbc6e826131d7ecec2eb3df49';
+    private static final string URL_ENDPOINT = 'https://api.openweathermap.org/data/2.5/weather?q=';
+    private static final string URL_ICON = 'https://openweathermap.org/img/wn/';
+    
+    public WeatherIntegrationControllerExtension(ApexPages.standardController sdtController){
+     	   this.accountRecord = (Account) sdtController.getRecord();
```

Comment:
In the constructor, you should initialize the private sdtController.
The property accountRecord will be initialized on the get. No need to do it in the constructor.